dotfiles are ignored

for instance if I have 
.gitignore
in my git status output. it can't be selected using PathPicker.

Thanks for checking out this ticket @pedrox. First, I'd like to make a clarification. When I say "dotfile" I mean "any file that begins with a period"[[1]].

> I believe dot files inside directory are not a problem.

I confirmed that PathPicker does **not** recognize dotfiles within a directory by adding the following case to `test.py` on the latest master branch commit.

```py
{
    'input': 'tmp/.gitignore',
    'match': True,
    'file': 'tmp/.gitignore',
    'num': 0
}
```
```bash
$ git log -n1 | head -n1
commit 7b0bef202b9f4783197459e26fe658f404b99188
$ python src/test.py
...
======================================================================
FAIL: testFileMatch (__main__.TestParseFunction)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "src/test.py", line 244, in testFileMatch
    self.checkFileResult(testCase)
  File "src/test.py", line 252, in checkFileResult
    testCase['input'])
AssertionError: Line "tmp/.gitignore" did not match any regex

----------------------------------------------------------------------
Ran 5 tests in 0.007s

FAILED (failures=2, errors=1)
```

> This is a duplicate of #7.

To my reading, issue #7 is about detecting files with no extensions (e.g. `Makefile`). While this issue is similar, I would suggest that this ticket is *not* a duplicate of #7 because it's quite possible to have a dotfile with an extension (e.g. `.env.production` or `.hidden.txt`).

[1]:http://en.wikipedia.org/wiki/Dot-file

--- a/src/parse.py
+++ b/src/parse.py
@@ -31,7 +31,7 @@ JUST_FILE = re.compile(
 FILE_NO_PERIODS = re.compile(''.join( (
     '(',
         # Recognized files starting with a dot followed by at least 3 characters
-        '(\.[a-zA-Z0-9\-_]{3,}[a-zA-Z0-9\-_\/]*)',
+        '((\/?([a-z.A-Z0-9\-_]+\/))?\.[a-zA-Z0-9\-_]{3,}[a-zA-Z0-9\-_\/]*)',
         # or
         '|',
         # Recognize files containing at least one slash

--- a/src/parse.py
+++ b/src/parse.py
@@ -27,7 +27,7 @@ HOMEDIR_REGEX = re.compile(
 OTHER_BGS_RESULT_REGEX = re.compile(
     '(\/?([a-z.A-Z0-9\-_]+\/)+[a-zA-Z0-9_.]{3,})[:-]{0,1}(\d+)')
 JUST_FILE = re.compile(
-    '([a-z.A-Z0-9\-_][a-z.A-Z0-9\-_]*\.[a-zA-Z]{1,10})(\s|$|:)+')
+    '([a-z.A-Z0-9\-_]+\.[a-zA-Z]{1,10})(\s|$|:)+')
 FILE_NO_PERIODS = re.compile(''.join( (
     '(',
         # Recognized files starting with a dot followed by at least 3 characters

--- a/src/parse.py
+++ b/src/parse.py
@@ -27,7 +27,7 @@ HOMEDIR_REGEX = re.compile(
 OTHER_BGS_RESULT_REGEX = re.compile(
     '(\/?([a-z.A-Z0-9\-_]+\/)+[a-zA-Z0-9_.]{3,})[:-]{0,1}(\d+)')
 JUST_FILE = re.compile(
-    '([a-zA-Z0-9\-_][a-z.A-Z0-9\-_]*\.[a-zA-Z]{1,10})(\s|$|:)+')
+    '([a-z.A-Z0-9\-_][a-z.A-Z0-9\-_]*\.[a-zA-Z]{1,10})(\s|$|:)+')
 FILE_NO_PERIODS = re.compile(
     '([a-z.A-Z0-9\-_\/]{1,}\/[a-zA-Z0-9\-_]{1,})(\s|$|:)+')
 

--- a/src/parse.py
+++ b/src/parse.py
@@ -28,8 +28,18 @@ OTHER_BGS_RESULT_REGEX = re.compile(
     '(\/?([a-z.A-Z0-9\-_]+\/)+[a-zA-Z0-9_.]{3,})[:-]{0,1}(\d+)')
 JUST_FILE = re.compile(
     '([a-z.A-Z0-9\-_][a-z.A-Z0-9\-_]*\.[a-zA-Z]{1,10})(\s|$|:)+')
-FILE_NO_PERIODS = re.compile(
-    '([a-z.A-Z0-9\-_\/]{1,}\/[a-zA-Z0-9\-_]{1,})(\s|$|:)+')
+FILE_NO_PERIODS = re.compile(''.join( (
+    '(',
+        # Recognized files starting with a dot followed by at least 3 characters
+        '(\.[a-zA-Z0-9\-_]{3,}[a-zA-Z0-9\-_\/]*)',
+        # or
+        '|',
+        # Recognize files containing at least one slash
+        '([a-z.A-Z0-9\-_\/]{1,}\/[a-zA-Z0-9\-_]{1,})',
+    ')',
+    # Regardless of the above case, here's how the file name should terminate
+    '(\s|$|:)+'
+) ))
 
 
 # Attempts to resolve the root directory of the

--- a/src/test.py
+++ b/src/test.py
@@ -39,6 +39,11 @@ fileTestCases = [{
     'file': './asd.txt',
     'num': 83
 }, {
+    'input': '.env.local',
+    'match': True,
+    'file': '.env.local',
+    'num': 0
+}, {
     'input': 'flib/asd/ent/berkeley/two.py-22',
     'match': True,
     'file': 'flib/asd/ent/berkeley/two.py',

--- a/src/test.py
+++ b/src/test.py
@@ -49,10 +49,27 @@ fileTestCases = [{
     'file': '.gitignore',
     'num': 0
 }, {
-    'input': '.bundle/config',
+    'input': 'tmp/.gitignore',
     'match': True,
-    'file': '.bundle/config',
+    'file': 'tmp/.gitignore',
     'num': 0
+}, {
+    'input': '.ssh/.gitignore',
+    'match': True,
+    'file': '.ssh/.gitignore',
+    'num': 0
+}, {
+    'input': '.ssh/known_hosts',
+    'match': True,
+    'file': '.ssh/known_hosts',
+    'num': 0
+
+# For now, don't worry about matching the following case perfectly,
+# simply because it's complicated.
+#}, {
+#    'input': '~/.ssh/known_hosts',
+#    'match': True,
+
 }, {
     # Arbitrarily ignore really short dot filenames
     'input': '.a',

--- a/src/test.py
+++ b/src/test.py
@@ -44,6 +44,20 @@ fileTestCases = [{
     'file': '.env.local',
     'num': 0
 }, {
+    'input': '.gitignore',
+    'match': True,
+    'file': '.gitignore',
+    'num': 0
+}, {
+    'input': '.bundle/config',
+    'match': True,
+    'file': '.bundle/config',
+    'num': 0
+}, {
+    # Arbitrarily ignore really short dot filenames
+    'input': '.a',
+    'match': False,
+}, {
     'input': 'flib/asd/ent/berkeley/two.py-22',
     'match': True,
     'file': 'flib/asd/ent/berkeley/two.py',

